### ISSUE TITLE
Node 14 circular dependency

### ISSUE THREAD

**Issue by MrWook** [nodkz/mongodb-memory-server#433]
## Versions

- NodeJS: 14.15.3
- mongodb-memory-server-*: 6.9.3 and 7.0.0-beta.14
- mongodb: 4.4.2
- system: MacOS, Linux
- Repo https://github.com/MrWook/mongodb-memory-server-circular-dependency

package: mongo-memory-server 

## What is the Problem?

We updated from node 12 to Node 14 and a warning is thrown with `Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency`

## Code Example

// jest-global.setup.js
```js
const { MongoMemoryServer } = require('mongodb-memory-server')

module.exports = async () => {
  const dbName = 'jestDatabaseName'

  global.mongodb = await MongoMemoryServer.create({
    instance: {
      dbName,
      port: 56490,
    },
  })
}
```

// jest-global.teardown.js
```js
module.exports = async () => {
  await global.mongodb.stop(true)
}
```

## Do you know *why* it happenes?

This seems like a general problem since node 14 for libraries like https://github.com/nodejs/node/issues/32987

The PR from node to throw this warning is https://github.com/nodejs/node/pull/29935
This means there should be a circular dependency in this library

Here is the whole stacktrace with `node --trace-warnings node_modules/.bin/jest --no-cache someTest.spec.ts`:

```
Determining test suites to run...(node:92425) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
at Object.get (internal/modules/cjs/loader.js:664:5)
at Object.<anonymous> (/someProjectPath/node_modules/mongodb/lib/operations/operation.js:4:38)
at Module._compile (internal/modules/cjs/loader.js:1063:30)
at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
at Object.newLoader (/someProjectPath/node_modules/pirates/lib/index.js:104:7)
at Module.load (internal/modules/cjs/loader.js:928:32)
at Function.Module._load (internal/modules/cjs/loader.js:769:14)
at Module.require (internal/modules/cjs/loader.js:952:19)
at require (internal/modules/cjs/helpers.js:88:18)
```

Which means the circular dependency is in `mongodb/lib/operations/operation.js:4:38` with `const MongoError = require('../core').MongoError;`

**Comment by MrWook** [comment]
I just saw this isn't even re right repo i'm sorry. Its the main mongodb repo

**Comment by MrWook** [comment]
Actually it is a combination from the global jest setup and `mongodb-memory-server`. If you use mongodb directly in the global jest setup to connect to a server everything works fine. If you create a server with this library it will throw the warning

I created a reproduction repo https://github.com/MrWook/mongodb-memory-server-circular-dependency

**Comment by hasezoey** [comment]
confirmed also happening when upgrading typegoose to `@beta` of this package

but i still have no clue what is causing this, but it dosnt seem to affect anything (yet?)

**Comment by hasezoey** [comment]
update: found an mongoose issue about it: https://github.com/Automattic/mongoose/issues/9900

TL;DR: its an mongodb driver issue, the driver needs to be updated to fix this error

**Comment by MichaelGrigoryan25** [comment]
I have also traced all the warnings and here's the result

```
node --trace-warnings src/index
(node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
    at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
    at Object.get (internal/modules/cjs/loader.js:664:5)
    at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
```

**Comment by MrWook** [comment]
I get it in 6.9.3 but like i said only when i use it in the jest global setup. It don't break anything on my end either it is just pretty scary to see :D 

Good thing it is something so simple, i guess it should be in the next mongodb release then

**Comment by hasezoey** [comment]
> I get it in 6.9.3 but like i said only when i use it in the jest global setup. It don't break anything on my end either it is just pretty scary to see :D

cant fix for now, because the version where it got fixed in the mongodb driver didnt release yet (and it dosnt seem like they will release it in an minor/patch version)

**Comment by hasezoey** [comment]
237177b82baf11610cf695bca9f3a81d8cbfb51f should fix this
